### PR TITLE
[Video] Move click handler out of video-manager

### PR DESCRIPTION
**Pull request by aghassemi**
Moving the click handler out of video-manager and creating a custom event type for it so that iframe-based players can also implement it.

We might also add an `user-interaction` event later but for autoplay we care about a `tap` anywhere on the player rather than a `user-interaction` which has a different meaning in my opinion (`user-interaction` implies playing, pausing, seeking, etc.. so a tap on the player that does not cause any action, won't be a `user-interaction`)

/to @mkhatib 
/cc @cramforce

### CODE REVIEW

File: builtins/amp-video.js [hunk 1]
```diff
@@ -108,6 +109,11 @@ export function installVideo(win) {
         this.video_.appendChild(child);
       });
 
+      // Dispatch a user tap event on click of the player.
+      listen(this.video_, 'click', () => {
+        this.element.dispatchCustomEvent(VideoEvents.USER_TAP);
```
Comment (mkhatib) [inline builtins/amp-video.js]: Add a testcase to make sure this is dispatched when video is clicked?

Comment (aghassemi) [inline builtins/amp-video.js]: done

File: test/functional/test-amp-video.js [hunk 2]
```diff
@@ -273,6 +275,19 @@ describe('amp-video', () => {
...
+  it('should dispatch VideoEvents.USER_TAP event when tapped', () => {
+    return getVideo({
+      src: 'video.mp4',
+      width: 160,
+      height: 90,
+    }).then(v => {
+      const video = v.querySelector('video');
+      const p = listenOncePromise(v, VideoEvents.USER_TAP);
+      video.dispatchEvent(new Event('click'));
+      return p;
```
Comment (mkhatib) [inline test/functional/test-amp-video.js]: Nice, this would basically timeout if it wasn't resolved?

Comment (aghassemi) [inline test/functional/test-amp-video.js]: Yeah, gets the Mocha's default 2 second timeout and fails the test (have tested it)

**Pull request merged**